# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/notification/Notification.cpp
# Commit: a8c5441fd896
# Full Hash: a8c5441fd89658dea45d939b1e930cf3d148fafd
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-01-26 21:47:24
# Description:
#   Bug 1873573 - Rename PromiseWorkerProxy::WorkerPromise() to GetWorkerPromise(). r=dom-worker-reviewers,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199323
# ==============================================================================

diff -r 533ecc686bac -r a8c5441fd896 dom/notification/Notification.cpp
--- a/dom/notification/Notification.cpp	Fri Jan 26 10:41:43 2024 +0000
+++ b/dom/notification/Notification.cpp	Fri Jan 26 10:48:56 2024 +0000
@@ -1700,7 +1700,11 @@
         mStrings(std::move(aStrings)) {}
 
   void WorkerRunInternal(WorkerPrivate* aWorkerPrivate) override {
-    RefPtr<Promise> workerPromise = mPromiseProxy->WorkerPromise();
+    RefPtr<Promise> workerPromise = mPromiseProxy->GetWorkerPromise();
+    // Once Worker had already started shutdown, workerPromise would be nullptr
+    if (!workerPromise) {
+      return;
+    }
 
     AutoTArray<RefPtr<Notification>, 5> notifications;
     for (uint32_t i = 0; i < mStrings.Length(); ++i) {